lib/spack/spack/util/compression.py

```diff
@@ -159,7 +163,12 @@ def _py_gunzip(archive_file):
 def _system_gunzip(archive_file):
     """Returns path to gunzip'd file
     Decompresses `.gz` compressed files via system gzip"""
-    decompressed_file = os.path.basename(strip_compression_extension(archive_file, "gz"))
+    archive_file_no_ext = strip_compression_extension(archive_file)
+    if archive_file_no_ext == archive_file:
+        # the zip file has no extension. On Unix gunzip cannot unzip onto itself
+        archive_file = archive_file + ".gz"
+        shutil.move(archive_file_no_ext, archive_file)
```

Comment:
I think this PR should apply the same level of care to `bz2`, `xz`, etc.: if the filename is missing the extension, we add it.

Reply:
The reason why we need these renames is to go around issues or bugs with the utilities we use to decompress. In one case, tar on windows complains that it cannot uncompress onto itself. In a second case, unzip on unix complains about a file with no extension. While we test all possible extensions, we don't hit other issues because the utilities we use don't have the same drawbacks. So, for simplicity I propose to not address the other extensions as there is no fundamental reason why these renames are needed. It is only to go around bugs in the utilities.

Reply:
Fair enough. I was inclined to say that these specific cases inform us of good general rules that - while not strictly required in all cases - might help us avoid similar complaints in the future (i.e. there's no harm in appending `.xz` and we may avoid some Windows/Unix-implementation-specific complaint in the future) - not something that has to be addressed here though (and I think there is definitely some value in knowing the actual tools with the problem).